core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -1299,6 +1291,10 @@ object Stream {
   def iterateEval[F[_],A](start: A)(f: A => F[A]): Stream[F,A] =
     emit(start) ++ eval(f(start)).flatMap(iterateEval(_)(f))
 
+  /** Allows to get current scope during evaluation of the stream **/
+  def getScope[F[_]]: Stream[F, Scope[F]] =
```

Comment:
Since we're making `Scope` part of the public api here, we'll need a `trait Scope[F[_]]` in the `fs2` package. We can keep the implementation of that trait in `fs2.internal` if that makes sense.

Reply:
ok makes sense will do